addons/sale/models/sale_order_line.py

```diff
@@ -1163,3 +1163,29 @@ def _convert_to_sol_currency(self, amount, currency):
                 round=False,
             )
         return amount
+
+    def action_add_from_catalog(self):
+        order_id = self.env.context.get('order_id')
+        sale_order = self.env['sale.order'].browse(order_id).exists()
```

Comment:
I do this because if the sale order is not saved yet then `self.order_id` points to `False`.

Reply:
Forget this, I do not even need `sale_order`.